server/handlers.go

```diff
@@ -243,6 +243,8 @@ func canContainsXSS(contentType string) bool {
 
 /* The preview handler will show a preview of the content for browsers (accept type text/html), and referer is not transfer.sh */
 func (s *Server) previewHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Vary", "Accept, X-Decrypt-Password")
```

Comment:
`previewHandler ` should not be sensible to `X-Decrypt-Password`

Reply:
indeed not event to `Accept`, or do I miss anything?

Reply:
I think the Vary header [should always be the same for the same URL](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary).

Reply:
>I think the Vary header [should always be the same for the same URL](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary).

I had this doubt just after leaving the comment

the for `previewHandler`, `getHandler` and `headHandler` we must set `Range, X-Decrypt-Password`
for `viewHandler` only `Accept`

Reply:
`viewHandler` is only for `/`, yes? I don't know if it should vary based on `X-Decrypt-Password`, only for the file route.
Is that okay?

Reply:
yes, `viewHandler` is only for `/`, no need to vary based on `X-Decrypt-Password`

indeed the file route must vary also for `Referer`

Reply:
Should be fixed in the latest commit

Reply:
sorry, file route should have `w.Header().Set("Vary", "Range, Referer, X-Decrypt-Password")`

Reply:
Sorry, I thought getHandler was the file route, do you mean zipHandler, tarHandler, tarGzHandler, etc should also have the vary header?

Reply:
these three ones:

https://github.com/dutchcoders/transfer.sh/pull/536/files#diff-961509ed3ff668f3172148150903514fb3394d6ff090d6db78d986875b13689fR246
https://github.com/dutchcoders/transfer.sh/pull/536/files#diff-961509ed3ff668f3172148150903514fb3394d6ff090d6db78d986875b13689fR1161
https://github.com/dutchcoders/transfer.sh/pull/536/files#diff-961509ed3ff668f3172148150903514fb3394d6ff090d6db78d986875b13689fR1255

currently they have `w.Header().Set("Vary", "Accept, Referer, X-Decrypt-Password")`
it should be `w.Header().Set("Vary", "Range, Referer, X-Decrypt-Password")`

response changes based on `Accept` only for `/` (`viewHandler`)
reponse for `/{token}/{filename}` changes `Range, Referer, X-Decrypt-Password`

`Range` and `X-Decrypt-Password` are used on `getHandler`
according to the `Referer` value we route to `previewHandler` or `getHandler`

finally `headHandler` should have the same `Vary` value since this should be the same based on URL and not URL+HTTP Method

Reply:
Got it, so sorry for the confusion!
Made the changes you requested